### PR TITLE
Add CLI commands for model gateway

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

 Depends on #8619

## What changes are proposed in this pull request?

Add CLI commands for model gateway.

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/gateway/cli.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import click
+
+
+@click.command("start-gateway", help="Start the MLflow Model Gateway service.")
```
[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/gateway/cli.py:4]: or can we use a separate sub-command like `mlflow gateway start`?

File: mlflow/models/cli.py [hunk 2]
```diff
@@ -275,3 +275,9 @@ def build_docker(model_uri, name, env_manager, mlflow_home, install_mlflow, enab
         install_mlflow=install_mlflow,
         enable_mlserver=enable_mlserver,
     )
+
+
+with contextlib.suppress(ImportError):
+    from mlflow.gateway.cli import start_gateway
+
+    commands.add_command(start_gateway)
```
Comment (harupy) [inline mlflow/models/cli.py:283]: `mlflow.gateway.cli` will import gateway-only dependencies such as `fastapi`.

File: mlflow/gateway/cli.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/gateway/cli.py:4]: +1 to the sub command. `mlflow models start-gateway` is too long.

[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/gateway/cli.py:4]: What options `mlflow gateway update` should have?

File: mlflow/gateway/cli.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    "--host",
+    default="127.0.0.1",
+    help="The network address to listen on (default: 127.0.0.1).",
+)
+@click.option(
+    "--port",
+    default=5000,
+    help="The port to listen on (default: 5000).",
+)
+def start_gateway(config_path: str, host: str, port: str):
```
[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/gateway/cli.py]: I think we should use the same approach as `mlflow server` to support uvicorn options.

[the remark this answers is not in the document]
Comment (BenWilson2) [reply mlflow/gateway/cli.py]: SGTM!

**Pull request closed without merging**